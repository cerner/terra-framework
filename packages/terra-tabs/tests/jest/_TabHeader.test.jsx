/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TabHeader from '../../src/common-tabs/_TabHeader';

describe('TabHeader', () => {
  test('should render a heading with provided title', () => {
    render((
      <TabHeader title="Test Title" />
    ));

    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toHaveAttribute('aria-level', '2');
    expect(screen.getByRole('heading')).toHaveAttribute('tabIndex', '-1');
    expect(screen.getByRole('heading')).toHaveTextContent('Test Title');

    expect(screen.getByText('Test Title', { selector: 'div' })).toHaveAttribute('aria-hidden', 'true');

    const wrapper = enzymeIntl.mountWithIntl(<TabHeader title="Test Title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
