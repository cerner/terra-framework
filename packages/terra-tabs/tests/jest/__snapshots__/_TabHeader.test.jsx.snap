// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabHeader should render a heading with provided title 1`] = `
<TabHeader
  intl={
    {
      "defaultFormats": {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": {},
      "formatters": {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  title="Test Title"
>
  <div
    className="header"
  >
    <VisuallyHiddenText
      aria-level="2"
      role="heading"
      tabIndex="-1"
      text="Test Title"
    >
      <span
        aria-level="2"
        className="visually-hidden-text"
        role="heading"
        tabIndex="-1"
      >
        Test Title
      </span>
    </VisuallyHiddenText>
    <div
      aria-hidden="true"
      className="title"
    >
      Test Title
    </div>
  </div>
</TabHeader>
`;
